lib/fcm.js

```diff
@@ -20,89 +21,96 @@ function FCM(serverKey) {
     };
 }
 
-module.exports = FCM;
+util.inherits(FCM, emitter);
 
 FCM.prototype.send = function(payload, CB) {
     var self = this;
     if (CB) this.once('sent', CB);
 
-    var operation = retry.operation();
-
-    payload = JSON.stringify(payload);
-
-    operation.attempt(function(currentAttempt) {
-        var headers = {
-            'Host': self.fcmOptions.host,
-            'Authorization': 'key=' + self.serverKey,
-            'Content-Type': 'application/json',
-            'Content-Length': new Buffer(payload).length
-        };
-
-        self.fcmOptions.headers = headers;
-
-        if (self.keepAlive) headers.Connection = 'keep-alive';
-
-        var request = https.request(self.fcmOptions, function(res) {
-            var data = '';
-
-
-            if (res.statusCode == 503) {
-                // If the server is temporary unavailable, the C2DM spec requires that we implement exponential backoff
-                // and respect any Retry-After header
-                if (res.headers['retry-after']) {
-                    var retrySeconds = res.headers['retry-after'] * 1; // force number
-                    if (isNaN(retrySeconds)) {
-                        // The Retry-After header is a HTTP-date, try to parse it
-                        retrySeconds = new Date(res.headers['retry-after']).getTime() - new Date().getTime();
+    return new Promise(function(resolve,reject){

```

Comment:
I think you shuld return promise if only callback is null or undefined.

alternative code: 
```js
if (cb) this.once('sent', cb);
else if (FCM.Promise) return new FCM.Promise(function (resolve, reject) {
  self.once('sent', function(err, response) {
    if (err) reject(err);
    else resolve(response);
  });
});
else throw new Error(...)
...

if (typeof Promise == 'function') FCM.Promise = Promise;
```

this will let user select what promise library they wanna use.